wodles/azure/azure-logs.py

```diff
@@ -602,11 +607,23 @@ def start_storage():
         logging.error("Storage: No parameters have been provided for authentication.")
         sys.exit(1)
 
+    # Authenticate
+    block_blob_service = BlockBlobService(account_name=name, account_key=key)
     try:
-        # Authenticate
-        block_blob_service = BlockBlobService(account_name=name, account_key=key)
+        # Disable max retry value before attempting to validate the credentials
+        old_retry_value = block_blob_service.retry
+        block_blob_service.retry = no_retry
+        # Verify credentials
+        block_blob_service.list_blobs(args.container, num_results=1)
+        # Restore the default max retry value
+        block_blob_service.retry = old_retry_value
         logging.info("Storage: Authenticated.")
+    except AzureException:
+        container_str = f"containers." if args.container == "*" else f"'{args.container}' container."
+        logging.error(f"Storage: Invalid credentials for accessing the {container_str}")
+        sys.exit(1)
```

Comment:
The same message is displayed when either wrong credentials were used or a wrong container. I think that if you used the [exists()](https://azure-storage.readthedocs.io/ref/azure.storage.blob.baseblobservice.html#azure.storage.blob.baseblobservice.BaseBlobService.exists) method instead, you could display the `Storage: Invalid credentials for accessing the {container_str}` when an exception is raised and a message like `Storage: The {container_str} does not exists` when the method returns False.

Reply:
The suggested change has been applied. Details in the comment below